extensions/sources/filebase/src/main/java/io/smallrye/ext/config/source/filebase/FileResourceWatcher.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.ext.config.source.filebase;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import lombok.extern.java.Log;
+
+/**
+ * Watching files for changes
+ * @author <a href="mailto:phillip.kruger@redhat.com">Phillip Kruger</a>
+ */
+@Log
+public class FileResourceWatcher {
+
+    private WatchService watcher;
+    
+    private final long pollInterval;
+    private final Reloadable reloadable;
+    
+    private final ScheduledExecutorService scheduledThreadPool = Executors.newSingleThreadScheduledExecutor();
+    
+    public FileResourceWatcher(Reloadable reloadable, long pollInterval){
+        this.reloadable = reloadable;
+        this.pollInterval = pollInterval;
+        
+        try {
+            this.watcher = FileSystems.getDefault().newWatchService();
+        } catch (IOException ex) {
+            log.log(Level.SEVERE, null, ex);
```

Comment:
This just leaves `watcher` as `null` without any other behavior - which is going to result in an endless stream of NPEs.

Reply:
Yup, Thanks. Now checking if watcher is initialized